File: src/gestorAplicacion/Profesor.java
package gestorAplicacion; //Falta especificar en qué sub-paquéte estará

import java.util.ArrayList;
import uiMain.BusquedaCursos;
import uiMain.Helpers;

public class Profesor extends Registro{
    private static final long serialVersionUID = 2L;
    private ArrayList<CursoProfesor> listaCursos = new ArrayList<CursoProfesor>();
    private Facultades facultad;
    private ArrayList<Double> calificaciones = new ArrayList<Double>();
    private double calificacion;

    public Profesor(String nombre, String correo, String nombreUsuario, String clave, String documento, ArrayList<CursoProfesor> listaCursos, Facultades facultad) {
        super(nombre, correo, nombreUsuario, clave, documento);
        this.listaCursos = listaCursos;
        this.facultad = facultad;
        this.calificacion = -1;
    }

    public double getCalificacion() {
      return calificacion;
    }

    public void setCalificacion(double calificacion) {
      this.calificacion = calificacion;
    }

    public ArrayList<CursoProfesor> getListaCursos() {
        return listaCursos;
    }

    public void setListaCursos(ArrayList<CursoProfesor> listaCursos) {
        this.listaCursos = listaCursos;
    }
    
    public void agegarCurso(CursoProfesor curso){
        this.listaCursos.add(curso);
    }

    public Facultades getFacultad() {
        return facultad;
    }

    public void setFacultad(Facultades facultad) {
        this.facultad = facultad;
    }


    public boolean agregarNota(Estudiante estudiante, Curso curso, int indiceNota, int indiceTipoNota, int nota) {
      if(!validarExistenciaCurso(curso)) return false;
      // TODO: mire si el estuainte existe entre los estudiantes de ese curso 
      // TODO: mire si la nota es valdia, crear metodo estatico para validar de forma general
      return true;
    }

    public boolean editarNota(Estudiante estudiante, Curso curso, int nota) {
      if(!validarExistenciaCurso(curso)) return false;
      // TODO: mire si el estuainte existe entre los estudiantes de ese curso 
      // TODO: mire si la nota es valdia, crear metodo estatico para validar de forma general
      return true;
    }
    
    public void agregarCurso(ArrayList<Curso> listaCursos, ArrayList<String> listaHorarios){ //Método único para crear profesores en el menú
        for(int x = 0; x<listaCursos.size(); x++){
            Curso curso = listaCursos.get(x);
            String horario = listaHorarios.get(x);
            CursoProfesor cp = new CursoProfesor(curso.getNombre(), curso.getId(), (short)curso.getCreditos(), curso.getNumeroParciales(), curso.getListaPorcentajes(), curso.getFacultad(), horario);
            this.listaCursos.add(cp);
            curso.agregarHorario(horario);
            curso.setCupos((short)(curso.getCupos()+5));
        }
    }

    private boolean validarExistenciaCurso(Curso curso) {
      for(CursoProfesor cursoActual: listaCursos) {
        if(cursoActual.getId() == curso.getId()) return true;
      }
      return false;
    }

    public void calificar(double valoracion) {
      calificaciones.add(valoracion);
      calificacion = Helpers.promedioLista(calificaciones);
    }
    
    @Override
    public void buscarCursos(){
        BusquedaCursos.buscarCursos();
    }

    public String toString() {
      return this.getNombre() + " (" + this.getCalificacion() + ")";
    }
}

File: PYTHON/src/gestorAplicacion/clasesDeUsuario/Profesor.py
from gestorAplicacion.clasesDeUsuario.Registro import Registro
from gestorAplicacion.clasesDeCurso.CursoProfesor import CursoProfesor

class Profesor(Registro):
    def __init__(self, nombre, correo, nombreUsuario, clave, documento, listaCursos, facultad):
        super().__init__(nombre, correo, nombreUsuario, clave, documento)
        self._listaCursos = listaCursos
        self._facultad = facultad
        self._calificaciones = []
        self._calificacion = -1
        Registro.agregarProfesor(self)

    def getCalificacion(self):
        return self._calificacion

    def setCalificacion(self, calificacion):
        self._calificacion = calificacion

    def getListaCursos(self):
        return self._listaCursos

    def setListaCursos(self, listaCursos):
        self._listaCursos = listaCursos

    def agegarCurso(self, curso):
        self._listaCursos.append(curso)

    def getFacultad(self):
        return self._facultad

    def setFacultad(self, facultad):
        self._facultad = facultad

    def agregarNota(self, estudiante, curso, indiceNota, indiceTipoNota, nota):
        if not self.validarExistenciaCurso(curso):
            return False
        # TODO: Check if the student exists among the students of that course
        # TODO: Check if the grade is valid, create a static method for general validation
        return True

    def editarNota(self, estudiante, curso, nota):
        if not self.validarExistenciaCurso(curso):
            return False
        # TODO: Check if the student exists among the students of that course
        # TODO: Check if the grade is valid, create a static method for general validation
        return True

    """ def agregarCurso(self, listaCursos, listaHorarios):
        for curso, horario in zip(listaCursos, listaHorarios):
            cp = CursoProfesor(curso.nombre, curso.id, curso.creditos, curso.numeroParciales, curso.listaPorcentajes, curso.facultad, horario)
            self._listaCursos.append(cp)
            curso.agregarHorario(horario)
            curso.cupos += 5 """

    def validarExistenciaCurso(self, curso):
        for cursoActual in self._listaCursos:
            if cursoActual._id == curso._id:
                return True
        return False

    def calificar(self, valoracion):
        self._calificaciones.append(valoracion)
        self.setCalificacion(round(sum(self._calificaciones)/len(self._calificaciones), 1))

    def buscarCursos(self):
        # Implement the logic for searching courses here
        pass

    def fueCalificado(self) -> bool :
        return False if self.getCalificacion() == -1 else True

    def __str__(self):
        return f"{self.getNombre()}\t{self.getCalificacion()}"